test/integration/nighthawk_test_server.py

```diff
@@ -283,6 +273,27 @@ def stop(self):
     os.remove(self._admin_address_path)
     return self.stopSubprocess()
 
+  @staticmethod
```

Comment:
I think my own preference would be for this to be in a utility file with supporting unit tests.  Understand this is out of scope for your PR, so happy with a TODO for now.